### PR TITLE
Hm setup android application

### PR DESCRIPTION
**Pull request by SanderVanAndel**
Implemented posts loading in the listview from the api
Implemented Chrome custom tab on click 

Before submitting your pull request, consider the following: Does your code...

- [x] Implement functionality
- [ ] Contain comments/documentation
- [x] Build without errors
- [ ] Contain and pass unit testing
- [ ] Conform to the Style Guide
- [x] Have no merge conflicts with ```master```

When you've checked all of the boxes above your code is up for review. If approved it can be merged with the ```master``` branch.

If you want feedback on your work before having finished your code, add ```[WIP]``` to the title of your PR.

### CODE REVIEW

File: android/src/main/java/nl/fhict/happynews/android/PostManager.java [hunk 1]
```diff
@@ -32,19 +34,22 @@ private PostManager() {
      * @param c context of the apps
      * @return
      */
-    public ArrayList<Post> getNewPosts(Context c){
-        final ArrayList<Post> newPosts = new ArrayList<>();
+    public void updatePosts(Context c) {
+        newPosts = new ArrayList<>();
```
Comment (SvenDub) [inline android/src/main/java/nl/fhict/happynews/android/PostManager.java]: What is the use of this variable? The posts are added to this list, but are never queried.

File: android/src/main/java/nl/fhict/happynews/android/PostManager.java [hunk 2]
```diff
@@ -32,19 +34,22 @@ private PostManager() {
...
         Ion.with(c)
                 .load(API_URL)
-                .as(new TypeToken<List<Post>>(){})
+                .as(new TypeToken<List<Post>>() {
+                })
                 .setCallback(new FutureCallback<List<Post>>() {
                     @Override
                     public void onCompleted(Exception e, List<Post> posts) {
```
Comment (SvenDub) [inline android/src/main/java/nl/fhict/happynews/android/PostManager.java]: We probably want to handle the case when the call fails. `Exception e` will be not null in that case.

File: android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java [hunk 3]
```diff
@@ -53,7 +48,13 @@ public int getCount() {
...
     public void updateData(ArrayList<Post> posts){
-        this.posts = posts;
-        this.notifyDataSetChanged();
+        if(posts != null){
+            this.posts = posts;
+            this.notifyDataSetChanged();
+        }
+    }
+
+    public Post getItem(int position) {
```
Comment (SvenDub) [inline android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java]: This method misses an `@Override` annotation.

File: android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java [hunk 4]
```diff
@@ -19,32 +19,27 @@
...
             convertView = LayoutInflater.from(getContext()).inflate(xml_type, parent, false);
-            postname = (TextView) convertView.findViewById(R.id.postTitle);
-            postname.setText(p.getTitle());
+            TextView postName = (TextView) convertView.findViewById(R.id.postTitle);
+            postName.setText(p.getTitle());
         }
-
-
         return convertView;
     }
```
Comment (SvenDub) [inline android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java]: This method misses an `@Override` annotation.

File: android/src/main/java/nl/fhict/happynews/android/Activitys/MainActivity.java [hunk 5]
```diff
@@ -1,49 +1,52 @@
...
+        postManager.updatePosts(getApplicationContext());
+
+        postList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                Post clickedPost = adapter.getItem(position);
+                if (!clickedPost.getUrl().isEmpty()) {
+                    Uri uri = Uri.parse(clickedPost.getUrl());
+                    CustomTabsIntent.Builder intentBuilder = new CustomTabsIntent.Builder();
+                    CustomTabsIntent customTabsIntent = intentBuilder.build();
```
Comment (DaanTul) [inline android/src/main/java/nl/fhict/happynews/android/Activitys/MainActivity.java]: maybe replace 2 lines with customtabsintent customtabsintent = new customtabsintent.builder().build();

File: android/src/main/java/nl/fhict/happynews/android/Activitys/MainActivity.java [hunk 6]
```diff
@@ -1,49 +1,52 @@
...
+                    customTabsIntent.launchUrl(MainActivity.this, uri);
+                } else {
+                    Toast.makeText(getApplicationContext(), "Link not found",
+                            Toast.LENGTH_LONG).show();
```
Comment (DaanTul) [inline android/src/main/java/nl/fhict/happynews/android/Activitys/MainActivity.java]: maybe short toast?

File: android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java [hunk 7]
```diff
@@ -53,7 +48,13 @@ public int getCount() {
...
 
     public void updateData(ArrayList<Post> posts){
```
Comment (DaanTul) [inline android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java]: add comments?

File: android/src/main/java/nl/fhict/happynews/android/Activitys/MainActivity.java (hunk 5 above)
Comment (SanderVanAndel) [inline android/src/main/java/nl/fhict/happynews/android/Activitys/MainActivity.java]: Seems more logical indeed

File: .idea/runConfigurations/android.xml [hunk 8]
```diff
@@ -1,33 +0,0 @@
-<component name="ProjectRunConfigurationManager">
```
Comment (SvenDub) [inline .idea/runConfigurations/android.xml]: We don't want to delete this.

File: android/src/main/java/nl/fhict/happynews/android/Activitys/MainActivity.java (hunk 6 above)
Comment (SanderVanAndel) [inline android/src/main/java/nl/fhict/happynews/android/Activitys/MainActivity.java]: Agree, changed it

File: android/src/main/java/nl/fhict/happynews/android/PostManager.java (hunk 1 above)
Comment (SanderVanAndel) [inline android/src/main/java/nl/fhict/happynews/android/PostManager.java]: Unnecessary indeed, will change it

File: android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java (hunk 7 above)
Comment (SanderVanAndel) [inline android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java]: Will update

File: .idea/runConfigurations/android.xml (hunk 8 above)
Comment (SanderVanAndel) [inline .idea/runConfigurations/android.xml]: I thought this was necessary for fixing the fixed listview sizing, which I needed for testing (saw this in Tom's commit (Fixed listview sizing in Main_Activity)). Tested it and it wasn't necessary, so i reverted it.

File: android/src/main/java/nl/fhict/happynews/android/PostManager.java (hunk 2 above)
Comment (SanderVanAndel) [inline android/src/main/java/nl/fhict/happynews/android/PostManager.java]: Agree, changed it

File: android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java (hunk 4 above)
Comment (SanderVanAndel) [inline android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java]: Added the needed overrides

**Pull request merged**